gdcdatamodel/validators/json_validators.py

```diff
@@ -57,3 +58,85 @@ def record_errors(self, entities):
                     )
                 entity.record_error(message, keys=keys)
             # additional validators go here
+
+    def allow_nulls(self, json_doc):
+        for key, value in json_doc.items():
+            if value == "null":
+                json_doc[key] = None
+            if value == "null" or value == None:
+                # dynamically allow "null" type for fields that are not required
+                if self.schemas.schema[json_doc["type"]]["required"]:
+                    if (
+                        key not in self.schemas.schema[json_doc["type"]]["required"]
+                        and key != "id"
+                        and key != "type"
+                    ):
+                        try:
+                            if (
+                                "null"
+                                not in self.schemas.schema[json_doc["type"]][key][
+                                    "type"
+                                ]
+                            ):
+                                if (
+                                    type(
+                                        self.schemas.schema[json_doc["type"]][key][
+                                            "type"
+                                        ]
+                                    )
+                                    != list
+                                ):
+                                    self.schemas.schema[json_doc["type"]][key][
+                                        "type"
+                                    ] = list(
+                                        self.schemas.schema[json_doc["type"]][key][
+                                            "type"
+                                        ]
+                                    )
+                                self.schemas.schema[json_doc["type"]][key]["type"] += [
+                                    "null"
+                                ]
+                        except:
```

Comment:
what kind of exception are we catching here? try to use `except <exception type>` as often as possible instead of catching all exceptions